packages/es-dev-server/src/middleware/plugin-transform.ts

```diff
@@ -18,12 +52,27 @@ export function createPluginTransformMiddlware(cfg: PluginServeMiddlewareConfig)
   }
 
   return async function pluginTransformMiddleware(context, next) {
+    let cached = false;
+    let cacheKey = '';
+    let cachedBody: string | undefined;
+
+    if (cfg.fileWatcher && cfg.cache) {
+      const result = await tryServeFromCache(cfg.cache, context);
+      ({ cached, cacheKey, cachedBody } = result);
+      context = result.context;
```

Comment:
What does reassigning the context do? Shouldn't this be like the old logic:

```
        ctx.body = cached.body;
        ctx.response.set(cached.headers);
        ctx.status = 200;
```

Reply:
Yep you're right, and that's what it's doing in tryServeFromCache I believe. I'm not sure why I am doing this here

Reply:
I removed it and it works :)